### PR TITLE
*: cut 0.29.0-rc.0

### PR DESCRIPTION
**Pull request by GiedriusS**
This looks like one of the bigger releases. I suggest releasing this now to make it smaller even though we have some exciting things in the pipeline.

We've been running this for a few days and haven't spotted any issues. I know that the Azure fixes work for at least one person so it would be really good to get this out.

Signed-off-by: Giedrius Statkevičius <giedrius.statkevicius@vinted.com>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,20 +8,21 @@ NOTE: As semantic versioning states all 0.y.z releases can contain breaking chan
...
+## [v0.29.0-rc.0](https://github.com/thanos-io/thanos/tree/release-0.29) - 2022.10.21
 
 ### Fixed
 - [#5642](https://github.com/thanos-io/thanos/pull/5642) Receive: Log labels correctly in writer debug messages.
 - [#5655](https://github.com/thanos-io/thanos/pull/5655) Receive: Fix recreating already pruned tenants.
 - [#5702](https://github.com/thanos-io/thanos/pull/5702) Store: Upgrade minio-go/v7 to fix panic caused by leaked goroutines.
 - [#5736](https://github.com/thanos-io/thanos/pull/5736) Compact: Fix crash in GatherNoCompactionMarkFilter.NoCompactMarkedBlocks.
 - [#5763](https://github.com/thanos-io/thanos/pull/5763) Compact: Enable metadata cache.
+- [#5759](https://github.com/thanos-io/thanos/pull/5759) Compact: fix missing duration log key.
+- [#5799](https://github.com/thanos-io/thanos/pull/5799) Query-Frontend: fixed sharding behaviour for vector matches. Now queries with sharding should work properly where the query looks like: `foo and without (lbl) bar`.
```
Comment (matej-g) [inline CHANGELOG.md:20]:
```suggestion
- [#5759](https://github.com/thanos-io/thanos/pull/5759) Compact: Fix missing duration log key.
- [#5799](https://github.com/thanos-io/thanos/pull/5799) Query Frontend: Fixed sharding behaviour for vector matches. Now queries with sharding should work properly where the query looks like: `foo and without (lbl) bar`.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,20 +8,21 @@ NOTE: As semantic versioning states all 0.y.z releases can contain breaking chan
...
 
 ### Added
+- [#5565](https://github.com/thanos-io/thanos/pull/5565) Receive: Allow remote write request limits to be defined per file and tenant.
```
Comment (matej-g) [inline CHANGELOG.md:23]:
```suggestion
- [#5565](https://github.com/thanos-io/thanos/pull/5565) Receive: Allow remote write request limits to be defined per file and tenant (experimental).
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -31,9 +32,13 @@ We use *breaking :warning:* to mark changes that are not backward compatible (re
 - [#5779](https://github.com/thanos-io/thanos/pull/5779) Objstore: Support specifying S3 storage class.
 - [#5741](https://github.com/thanos-io/thanos/pull/5741) Query: add metrics on how much data is being selected by downstream Store APIs.
 - [#5673](https://github.com/thanos-io/thanos/pull/5673) Receive: Reload tenant limit configuration on file change.
+- [#5749](https://github.com/thanos-io/thanos/pull/5749) Query-Frontend: added small LRU cache to cache query analysis results.
```
Comment (matej-g) [inline CHANGELOG.md:35]:
```suggestion
- [#5749](https://github.com/thanos-io/thanos/pull/5749) Query Frontend: Added small LRU cache to cache query analysis results.
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -31,9 +32,13 @@ We use *breaking :warning:* to mark changes that are not backward compatible (re
...
 
 ### Changed
 
+- [#5738](https://github.com/thanos-io/thanos/pull/5738) Global: replace crypto/sha256 with minio/sha256-simd to make it faster.
```
Comment (matej-g) [inline CHANGELOG.md:39]: To make what faster? How? :smile: :racing_car: 
```suggestion
- [#5738](https://github.com/thanos-io/thanos/pull/5738) Global: Replace `crypto/sha256` with `minio/sha256-simd` to make it faster.
```

**Review by matej-g — APPROVED** [review]
Biggest release yet? :eyes: 



Couple of nits, but looks good otherwise :beers:

File: CHANGELOG.md (hunk 4 above)
Comment (squat) [reply CHANGELOG.md:39]: It enables parallel computation via SIMD. I guess we can mention that

Comment (GiedriusS) [reply CHANGELOG.md]: Removed `it` here and wrote specifically that this makes hash calculation faster in reloader and metadata packages.

**Review by bwplotka — APPROVED** [review]
💪🏽

**Pull request merged**